src/SimpleSoft.Database.EFCore/EFCoreTransaction.cs

```diff
@@ -0,0 +1,118 @@
+﻿using Microsoft.EntityFrameworkCore.Storage;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SimpleSoft.Database
+{
+    /// <inheritdoc />
+    public class EFCoreTransaction : ITransaction
+    {
+        private readonly EFCoreContextContainer _container;
+        private IDbContextTransaction _transaction;
+        private bool _disposed;
+
+        /// <summary>
+        /// Creates a new instance
+        /// </summary>
+        /// <param name="container"></param>
+        public EFCoreTransaction(EFCoreContextContainer container)
+        {
+            _container = container;
+        }
+
+        #region IDisposable
+
+        /// <inheritdoc />
+        ~EFCoreTransaction() => Dispose(false);
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting
+        /// unmanaged resources.
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+
+#if NETSTANDARD2_1
+                DisposeAsync()
+                    .ConfigureAwait(false)
+                    .GetAwaiter()
+                    .GetResult();
+#else
+                _transaction?.Dispose();
+#endif
+            }
+
+            _transaction = null;
+            _disposed = true;
+        }
+
+#if NETSTANDARD2_1 
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting
+        /// unmanaged resources asynchronously.
+        /// </summary>
+        /// <returns>A completed task of this operation</returns>
+        public virtual async ValueTask DisposeAsync()
+        {
+            if (_disposed)
+                return;
+
+            await _transaction.DisposeAsync();
```

Comment:
The `_transaction` field should be set to `null` after and that possibility must also be taken into account when disposing:

```csharp
public virtual async ValueTask DisposeAsync()
{
    if (_disposed)
        return;
    
    if(_transaction != null)
        await _transaction.DisposeAsync();
    
    _transaction = null;
    _disposed = true;
}
```